src/distilabel/pipeline/_dag.py

```diff
@@ -749,3 +750,141 @@ def from_dict(cls, data: Dict[str, Any]) -> "DAG":
+            for step in dump["steps"]
+        }
+        connections = dump["connections"]
+
+        step_outputs = {}
+        for step in dump["steps"]:
+            try:
+                step_outputs[step["name"]] = self.get_step(step["name"])[
+                    STEP_ATTR_NAME
+                ].get_outputs()
+            except AttributeError:
+                step_outputs[step["name"]] = {"dynamic": True}
+
+        # Add Argilla and Distiset steps to the graph
+        leaf_steps = self.leaf_steps
+        for idx, leaf_step in enumerate(leaf_steps):
+            if "to_argilla" in leaf_step:
+                connections.append({"from": leaf_step, "to": [f"to_argilla_{idx}"]})
+                step_name_to_class[f"to_argilla_{idx}"] = "Argilla"
+                step_outputs[leaf_step] = {"records": True}
+            else:
+                connections.append({"from": leaf_step, "to": [f"distiset_{idx}"]})
+                step_name_to_class[f"distiset_{idx}"] = "Distiset"
+
+        # Create a set of all steps in the graph
+        all_steps = {con["from"] for con in connections} | {
+            to_step for con in connections for to_step in con["to"]
+        }
+
+        # Create a mapping of step outputs
+        step_output_mappings = {
+            step["name"]: {
+                k: v
+                for k, v in {
+                    **{output: output for output in step_outputs[step["name"]]},
+                    **step["step"]["output_mappings"],
+                }.items()
+                if list(
+                    dict(
+                        {
+                            **{output: output for output in step_outputs[step["name"]]},
+                            **step["step"]["output_mappings"],
+                        }.items()
+                    ).values()
+                ).count(v)
+                == 1
+                or k != v
+            }
+            for step in dump["steps"]
+        }
+
+        return (
+            all_steps,
+            step_name_to_class,
+            connections,
+            step_outputs,
+            step_output_mappings,
+        )
+
+    def _to_mermaid_image(self, graph_styled: str) -> str:
+        """Converts a Mermaid graph to an image using the Mermaid Ink service.
+
+        Parameters:
+            graph_styled: The Mermaid graph to convert to an image.
+
+        Returns:
+            The image content.
+        """
+        base64_string = base64.b64encode(graph_styled.encode("ascii")).decode("ascii")
+        url = f"https://mermaid.ink/img/{base64_string}?type=png"
+
+        try:
+            response = requests.get(url, timeout=10)
+            response.raise_for_status()
+            return response.content
+        except requests.RequestException as e:
+            raise ValueError(
+                "Error accessing https://mermaid.ink/. See stacktrace for details."
+            ) from e
```

Comment:
This method could be placed as a standalone function outside of the `_DAG` as it doesn't need it.